src/levenberg.jl

```diff
@@ -228,21 +228,20 @@ end
 
 function jacobian_caches(alg::LevenbergMarquardt, f, u, p, ::Val{true})
     uf = JacobianWrapper(f, p)
-    J = ArrayInterface.undefmatrix(u)
+
+    du1 = zero(u)
+    du2 = zero(u)
+    tmp = zero(u)
+    J, jac_config = build_jac_and_jac_config(alg, f, uf, du1, u, tmp, du2)
 
     linprob = LinearProblem(J, _vec(zero(u)); u0 = _vec(zero(u)))
     weight = similar(u)
-    recursivefill!(weight, false)
+    # Q: Setting this to false leads to residual = 0 in GMRES?
+    recursivefill!(weight, true)
```

Comment:
```suggestion
    recursivefill!(weight, true)
```

yes, what was a bug, weight matrix should be identity by default not the zero matrix.